drivers/wifi/esp8266/wifi_esp8266.c

```diff
@@ -0,0 +1,1705 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wifi_esp8266, CONFIG_WIFI_LOG_LEVEL);
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+#include <net/net_offload.h>
+#include <gpio.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/net_context.h>
+#include <net/net_offload.h>
+#include <net/wifi_mgmt.h>
+#include <net/net_ip.h>
+
+#include "ipv4.h"
+#if defined(CONFIG_NET_UDP)
+#include "udp_internal.h"
+#endif
+
+#include "../../modem/modem_receiver.h"
+
+#define ESP8266_MAX_CONNECTIONS 5
+#define BUF_ALLOC_TIMEOUT K_SECONDS(1)
+#define MDM_MAX_DATA_LENGTH     1600
+
+struct cmd_handler {
+	const char *cmd;
+	u16_t cmd_len;
+	void (*func)(struct net_buf **buf, u16_t len);
+	bool skip;
+};
+
+#define CMD_HANDLER_NO_SKIP(cmd_, cb_) {	    \
+		.cmd = cmd_,			    \
+		.cmd_len = (u16_t)sizeof(cmd_) - 1, \
+		.func = on_cmd_ ## cb_,		    \
+		.skip = false			    \
+}
+
+#define CMD_HANDLER(cmd_, cb_) {		    \
+		.cmd = cmd_,			    \
+		.cmd_len = (u16_t)sizeof(cmd_) - 1, \
+		.func = on_cmd_ ## cb_,		    \
+		.skip = true			    \
+}
+
+static u8_t mdm_recv_buf[MDM_MAX_DATA_LENGTH];
+
+/* net bufs */
+NET_BUF_POOL_DEFINE(esp8266_recv_pool, 12, 128, 0, NULL);
+static struct k_work_q esp8266_workq;
+
+/* RX thread structures */
+K_THREAD_STACK_DEFINE(esp8266_rx_stack, 1024);
+
+/* RX thread work queue */
+K_THREAD_STACK_DEFINE(esp8266_workq_stack, 1024);
+
+struct k_thread esp8266_rx_thread;
+
+K_SEM_DEFINE(sock_sem, 1, 1);
+K_MUTEX_DEFINE(dev_mutex);
+
+struct socket_data {
+	struct net_context      *context;
+	sa_family_t family;
+	enum net_sock_type type;
+	enum net_ip_protocol ip_proto;
+	net_context_send_cb_t send_cb;
+	net_context_recv_cb_t recv_cb;
+
+
+	struct k_work recv_cb_work;
+	void *recv_user_data;
+
+	struct sockaddr src;
+	struct sockaddr dst;
+	struct net_pkt          *rx_pkt;
+	int ret;
+	int conn_id;
+	/** semaphore */
+	int connected;
+	int read_error;
+};
+
+struct conn_data {
+	struct socket_data *sock;
+};
+
+struct esp8266_data {
+	struct net_if *iface;
+	struct in_addr ip;
+	struct in_addr gw;
+	struct in_addr netmask;
+	u8_t mac[6];
+
+	/* modem stuff */
+	struct mdm_receiver_context mdm_ctx;
+	int last_error;
+
+	struct k_delayed_work ip_addr_work;
+
+	/* semaphores */
+	struct k_sem response_sem;
+
+	/* max sockets */
+	struct socket_data socket_data[ESP8266_MAX_CONNECTIONS];
+	u8_t sock_map;
+	struct conn_data conn_data[ESP8266_MAX_CONNECTIONS];
+	u8_t conn_map;
+
+	/* variables required by tcp server */
+	net_tcp_accept_cb_t accept_cb;
+	void *accept_data;
+	struct socket_data server_sock;
+
+	/* wifi scan callbacks */
+	scan_result_cb_t wifi_scan_cb;
+
+	int sending_data;
+
+	int data_id;
+	int data_len;
+
+	/* device management */
+	struct device *uart_dev;
+#ifdef CONFIG_WIFI_ESP8266_HAS_ENABLE_PIN
+	struct device *gpio_dev;
+#endif
+};
+
+static struct esp8266_data foo_data;
```

Comment:
Can you look at naming conventions; foo_data should be changed. You could use something more applicable ie modem_data or esp_data.

Billy..